### DISCUSSION TITLE
v18.1.5

### DISCUSSION THREAD

**Discussion by arturovt** [ngxs/store#2253] [Releases]
## What's Changed
* fix(store): prevent writing to state once action handler is unsubscribed by @arturovt in https://github.com/ngxs/store/pull/2231
* perf(store): replace `instanceof Function` with `typeof` by @arturovt in https://github.com/ngxs/store/pull/2247
* refactor(store): use `Object.is` as default equality check by @arturovt in https://github.com/ngxs/store/pull/2245
* refactor(router-plugin): mark selectors as pure by @arturovt in https://github.com/ngxs/store/pull/2248
* refactor(storage-plugin): mark engine tokens as pure by @arturovt in https://github.com/ngxs/store/pull/2249
* refactor(store): tree-shake internal state tokens by @arturovt in https://github.com/ngxs/store/pull/2246
* chore: release v18.1.5 by @arturovt in https://github.com/ngxs/store/pull/2252

**Full Changelog**: https://github.com/ngxs/store/compare/v18.1.4...v18.1.5

<hr /><em>This discussion was created from the release <a href='https://github.com/ngxs/store/releases/tag/v18.1.5'>v18.1.5</a>.</em>

**Comment by kleitensV** [comment]
Hi! I encountered an issue with this version.  In version 18.1.4 it's working fine.
When you use two receivers and calling one method from the other.

@Receiver()
  public static async firstMethod(context: StateContext<IFGroupDetailsModalState>
  ){
  
  await lastValueFrom (this.api.someApi());
   context.setState(produce(draft=>{
      draft.data = newData
   }))
   this.secondMethod(context)
  }
  
  @Receiver()
  public static async secondMethod(context: StateContext<IFGroupDetailsModalState>
  ){
  await lastValueFrom (this.api.someOtherApi());
  context.setState(produce(draft=>{
      draft.example=newExample
   }))
  }
  
  The setState produce in the secondMethod is not triggered. It's skipped.
  You cannot add breakpoints and no error is triggered.
  
  It should be possible to update state calling these methods one from the other?
